controllers/imagerepository_controller.go

```diff
@@ -231,6 +231,51 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	return ctrl.Result{}, nil
 }
 
+func (r *ImageRepositoryReconciler) AddNotifications(ctx context.Context, imageRepository *imagerepositoryv1alpha1.ImageRepository) ([]imagerepositoryv1alpha1.NotificationStatus, error) {
+	log := ctrllog.FromContext(ctx).WithName("ConfigureNotifications")
+	log.Info("Configuring notifications", "Name", imageRepository.Name, "Namespace", imageRepository.Namespace)
+
+	if imageRepository.Spec.Notifications == nil {
+		// No notifications to configure
+		return nil, nil
+	}
+
+	notificationStatus := []imagerepositoryv1alpha1.NotificationStatus{}
+
+	for _, notification := range imageRepository.Spec.Notifications {
+		log.Info("Creating notification in Quay", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method)
+		quayNotification, err := r.QuayClient.CreateNotification(
```

Comment:
It will return an error if this function is called second time.
See how, for example, `CreateRobotAccount` is implemented.